docs/user/alerting/create-and-manage-rules.asciidoc

```diff
@@ -145,17 +153,29 @@ Some cases exist where the variable values will be "escaped", when used in a con
 
 Mustache also supports "triple braces" of the form `{{{variable name}}}`, which indicates no escaping should be done at all.  Care should be used when using this form, as it could end up rendering the variable content in such a way as to make the resulting parameter invalid or formatted incorrectly.
 
+[float]
+[[defining-rules-actions-variable-context]]
+===== Action variable context
+
 Each rule type defines additional variables as properties of the variable `context`.  For example, if a rule type defines a variable `value`, it can be used in an action parameter as `{{context.value}}`.  
 
-For diagnostic or exploratory purposes, action variables whose values are objects, such as `context`, can be referenced directly as variables.  The resulting value will be a JSON representation of the object.  For example, if an action parameter includes `{{context}}`, it will expand to the JSON representation of all the variables and values provided by the rule type.
+For diagnostic or exploratory purposes, action variables whose values are objects, such as `context`, can be referenced directly as variables. The resulting value will be a JSON representation of the object.  For example, if an action parameter includes `{{context}}`, it will expand to the JSON representation of all the variables and values provided by the rule type. To see all variables, use `{{.}} `.
 
-You can attach more than one action. Clicking the *Add action* button will prompt you to select another rule type and repeat the above steps again.
+For situations where your rule response returns arrays of data, you can loop through the `context` by
 
-[NOTE]
-==============================================
-Actions are not required on rules. You can run a rule without actions to
-understand its behavior, then <<action-settings,configure actions>> later.
-==============================================
+[source]
+--------------------------------------------------
+{{#context}}{{.}}{{/context}}
+--------------------------------------------------
+
+For example, looping through search result hits may appear
```

Comment:
This sentence doesn't read correctly to me.  I wonder if we should show the "data" coming in (`{context: { hits: [ { _source: { message, ...` kinda thing, and then also show how that would actually be rendered?

Reply:
@pmuellr as a MVP, we could update the text to better reflect that the response will be dynamic and users will need to use `{{.}}` to see what's available to them. After we could say "for this example", but I'd hate to delay the "good" for the "perfect".

Reply:
WORKSFORME